util/dvsim/SynCfg.py

```diff
@@ -0,0 +1,372 @@
+        # empty) fields
+        #
+        # results = {
+        #     "tool": "dc",
+        #     "top" : <name of toplevel>,
+        #
+        #     "messages": {
+        #         "flow_errors"      : [],
+        #         "flow_warnings"    : [],
+        #         "analyze_errors"   : [],
+        #         "analyze_warnings" : [],
+        #         "elab_errors"      : [],
+        #         "elab_warnings"    : [],
+        #         "compile_errors"   : [],
+        #         "compile_warnings" : [],
+        #     },
+        #
+        #     "timing": {
+        #         # per timing group (ususally a clock domain)
+        #         # in nano seconds
+        #         <group>  : {
+        #             "tns"    : <value>,
+        #             "wns"    : <value>,
+        #             "period" : <value>,
+        #         ...
+        #         }
+        #     },
+        #
+        #     "area": {
+        #         # gate equivalent of a NAND2 gate
+        #         "ge"     : <value>,
+        #
+        #         # summary report, in GE
+        #         "comb"   : <value>,
+        #         "buf"    : <value>,
+        #         "reg"    : <value>,
+        #         "macro"  : <value>,
+        #         "total"  : <value>,
+        #
+        #         # hierchical report of first submodule level
+        #         "instances" : {
+        #             <name> : {
+        #               "comb"  : <value>,
+        #               "buf"   : <value>,
+        #               "reg"   : <value>,
+        #               "macro" : <value>,
+        #               "total" : <value>,
+        #             },
+        #             ...
+        #         },
+        #     },
+        #
+        #     "power": {
+        #         "net"  : <value>,
+        #         "int"  : <value>,
+        #         "leak" : <value>,
+        #     },
+        #
+        #     "units": {
+        #         "voltage"     : <value>,
+        #         "capacitance" : <value>,
+        #         "time"        : <value>,
+        #         "dynamic"     : <value>,
+        #         "static"      : <value>,
+        #     }
+        # }
+        #
+        # note that if this is a master config, the results will
+        # be generated using the _gen_results_summary function
+        # '''
+
+        def _create_entry(val, norm=1.0, total=None, perctag="%"):
+            """
+            Create normalized entry with an optional
+            percentage appended in brackets.
+            """
+            if val is not None and norm is not None:
+                if total is not None:
+                    perc = float(val) / float(total) * 100.0
+                    entry = "%2.1f %s" % (perc, perctag)
+                else:
+                    value = float(val) / norm
+                    entry = "%2.1f" % (value)
+            else:
+                entry = "--"
+
+            return entry
+
+        self.result = {}
```

Comment:
You might want to make this OrderedDict to supprt py 3.5.x.

Reply:
this does not iterate over these results, hence this should be safe.